lib/Controller/DocumentController.php

```diff
@@ -426,7 +426,7 @@ private function handleDocIndex($fileId, $shareToken, $currentUser) {
 
 		// Update permissions
 		$permissions = $doc['permissions'];
-		if (!($doc['action'] === 'edit')) {
+		if (!($doc['action'] === 'edit') && !($doc['action'] === 'view_comment')) {
```

Comment:
it is **&&** or **||** ? hm

Reply:
`&&` is correct 🙂 

The action for PDFs is `view_comment`, and we want to include the update permission in this case. With `||` we would run into the if-block which would remove this permission.

Reply:
As @JammingBen pointed out, the line:
`$permissions & ~\OCP\Constants::PERMISSION_UPDATE;`
removes the PERMISSION_UPDATE from the pemissions var